File: examples/js/src/templates/006-mail-support.ts
import { ChatAnthropic } from "@langchain/anthropic";
import * as z from "zod";

const mailSchema = z.object({
  subject: z.string("The subject of the mail"),
  message: z.string("The message of the mail"),
});

const model = new ChatAnthropic({
  model: "claude-opus-4-8",
}).withStructuredOutput(mailSchema);

async function answerSupportMail(subject: string, message: string) {
  const response = await model.invoke([
    {
      role: "system",
      content: `You are a helpful assistant that answers support mails. 
        Please create a response for the given mail.`,
    },
    {
      role: "user",
      content: `Subject: ${subject}\nMessage: ${message}`,
    },
  ]);
  console.log(response.subject);
  console.log(response.message);
}

answerSupportMail(
  "Cancel Subscription",
  "Hi, Cancel my subscription of the letter app. Now. Today. Immediately! No Regards, Donald",
);

File: examples/python/src/templates/006_mail_support.py
import asyncio
from langchain_anthropic import ChatAnthropic
from pydantic import BaseModel


class MailResponse(BaseModel):
    subject: str
    message: str


model = ChatAnthropic(model="claude-opus-4-8").with_structured_output(MailResponse)


async def answer_support_mail(subject: str, message: str):
    response = await model.ainvoke([
        {
            "role": "system",
            "content": "You are a helpful assistant that answers support mails. Please create a response for the given mail.",
        },
        {
            "role": "user",
            "content": f"Subject: {subject}\nMessage: {message}",
        },
    ])
    print(response.subject)
    print(response.message)


asyncio.run(answer_support_mail(
    "Cancel Subscription",
    "Hi, Cancel my subscription of the letter app. Now. Today. Immediately! No Regards, Donald",
))
